src/definitionProvider.ts

```diff
@@ -145,4 +142,142 @@ export class DataformRequireDefinitionProvider implements vscode.DefinitionProvi
+                const position = await getPostionOfSourceDeclaration(filePathUri, variableName);
+                if (position){
+                    return new vscode.Location(filePathUri, position);
+                };
+            }
+        }
+    } catch (error) {
+        console.error(`Error reading includes directory: ${error}`);
+    }
+    // If not found in includes directory, check if it is imported
+    const importedModules = getImportedModules(document);
+    const importedModule = importedModules.find(module => module.module === jsFileName);
+    if (importedModule) {
+        const filePath = path.join(workspaceFolder, importedModule.path);
+        const filePathUri = vscode.Uri.file(filePath);
+        const position = await getPostionOfSourceDeclaration(filePathUri, variableName);
+        if (position){
+            return new vscode.Location(filePathUri, position);
+        };
+    }
+
+    return undefined;
+}
+
+
+export function findModuleDefinition(
+        document: vscode.TextDocument,
+        workspaceFolder: string,
+        searchTerm: string
+    ) {
+        const includesPath = path.join(workspaceFolder, 'includes');
+        const importedModules = getImportedModules(document);
+        const importedModule = importedModules.find(module => module.module === searchTerm);
+        if (importedModule) {
+            const locationPath = path.join(workspaceFolder, importedModule.path);
+            return getLocationFromPath(locationPath);
+        }
+
+        const files = fs.readdirSync(includesPath);
+        for (const file of files) {
+            const filePath = path.join(includesPath, file);
+            const stat = fs.statSync(filePath);
+            if (!stat.isFile()){
+                continue;
+            }
+            if (file === `${searchTerm}.js`) {
+                return getLocationFromPath(filePath);
+            }
+        }
+        return undefined;
+}
+export class DataformJsDefinitionProvider implements vscode.DefinitionProvider {
+    async provideDefinition(
+        document: vscode.TextDocument,
+        position: vscode.Position,
+        token: vscode.CancellationToken
+    ): Promise<vscode.Location | undefined> {
+        const wordRange = document.getWordRangeAtPosition(position);
+        if(!wordRange){
+            return undefined;
+        }
+        const searchTerm = document.getText(wordRange);
+        const workspaceFolder = getWorkspaceFolder();
+
+        // Return if no workspace folder is available
+        if (!workspaceFolder) {
+            return undefined;
+        }
+        const line = document.lineAt(position.line).text;
+        // const start = wordRange.start.character;
+        // const end = wordRange.end.character;
+
+        // This will often not be inside ${}, rather it will be inside js {}
+        if (line.includes('require("')){
+            return findModuleDefinition(document, workspaceFolder, searchTerm);
+        }
```

Comment:
You're probably right, my thougt was that it's unlikely that it will be something else since the line contains "require(".

A check that only return if the value is not undefined is probably good enough?

Reply:
> A check that only return if the value is not undefined is probably good enough?

yes, that would suffice, thanks

Reply:
fixed